web/frontend/src/components/audio-input/index.svelte

```diff
@@ -15,7 +17,7 @@ let isOn = false;
 const toggleExpand = async () => {
   isOn = !isOn;
 
-  const streams = new StreamClient(client);
+  const streams = new StreamClient($client);
 
   streams.on('track', (event) => {
```

Comment:
Not new code, but should this do any work to remove the `track` listener once it's fired? IIRC, the `streams` events are bit aggressive/unintuitive in the TypeScript SDK, with a single event emitter singleton that will emit a `track` event on all clients, regardless of which client the track is for

Reply:
I added a cleanup function for when this component this is destroyed / disconnected, but that's all I'm comfortable with now. I definitely agree with the confusion, that event emitter was added very quickly by yours truly to circumvent an even worse scenario in the past where the TS SDK would create an entirely manage an html audio / video element, making working with it and reactivity a true pain in the butt. This should definitely be fixed!

cc @maximpertsov (apologies for all the pings, hopefully this feedback is interesting / useful!)

Reply:
Of course, all of this feedback is great.

> with a single event emitter singleton that will emit a track event on all clients, regardless of which client the track is for

yes we need to partition events by streams - there's a TODO for this that I will turn into a ticket for if there isn't one already.